lib/charms/vault_k8s/v0/vault_kv.py

```diff
@@ -447,47 +400,40 @@ def remove_unit_credentials(self, relation: ops.Relation, nonce: str | Iterable[
         for n in nonce:
             credentials.pop(n, None)
 
-        relation.data[self.charm.app]["credentials"] = json.dumps(credentials, sort_keys=True)
+        self.juju_facade.set_app_relation_data(
+            relation_name=self.relation_name,
+            relation=relation,
+            data={"credentials": json.dumps(credentials, sort_keys=True)},
+        )
 
     def get_credentials(self, relation: ops.Relation) -> dict:
-        """Get the unit credentials from the relation."""
-        return json.loads(relation.data[self.charm.app].get("credentials", "{}"))
-
-    def get_outstanding_kv_requests(self, relation_id: int | None = None) -> List[KVRequest]:
-        """Get the outstanding requests for the relation."""
-        outstanding_requests: List[KVRequest] = []
-        kv_requests = self.get_kv_requests(relation_id=relation_id)
-        for request in kv_requests:
-            if not self._credentials_issued_for_request(
-                nonce=request.nonce, relation_id=request.relation_id
-            ):
-                outstanding_requests.append(request)
-        return outstanding_requests
+        """Get the unit credentials from the relation and load it as a dict."""
+        return json.loads(
+            self.juju_facade.get_app_relation_data(
+                relation_name=self.relation_name,
+                relation=relation,
+            ).get("credentials", "{}")
+        )
```

Comment:
The docstring contradicts the method content, is it the unit or the app relation data?

Reply:
app, added.